samza-core/src/main/java/org/apache/samza/container/grouper/task/TaskAssignmentManager.java

```diff
@@ -91,8 +98,8 @@ public TaskAssignmentManager(Config config, MetricsRegistry metricsRegistry, Ser
    */
   public Map<String, String> readTaskAssignment() {
```

Comment:
Does this need to be updated to return the task mode too? Alternatively, do we need a separate getter for TaskMode?

Reply:
yes, I have a separate PR that introduces and uses a separate getter for TaskModes.